### DISCUSSION TITLE
Omitting taxonomies and terms from search engine indexing

### DISCUSSION THREAD

**Discussion by danielfdickinson** [adityatelange/hugo-PaperMod#529] [Ideas]
I wonder how folks feel about modifying ``head.html`` to have

```l
{{- if and (hugo.IsProduction | or (eq .Site.Params.env "production")) (not .Data.Terms) (not .Data.Pages) }}
<meta name="robots" content="index, follow">
{{- else }}
<meta name="robots" content="noindex, nofollow">
{{- end }}

```
So that taxonomies and terms do not get indexed by search engines. It seems to me these are not normally 'real' content pages and SEO analysis complains about the lack of description, so it seem preferrable to simply not index them. The lack of description can be dealt with, but it seems to me it's not all that useful to have these pages indexed anyway.

**Comment by adityatelange** [comment]
`robots.txt` is made for that https://www.cloudflare.com/learning/bots/what-is-robots.txt/

**Reply by adityatelange** [reply]
Add it in`layouts/robots.txt`

**Reply by danielfdickinson** [reply]
I don't know why I didn't think of that. Thanks.

**Comment by thiyagu-7** [comment]
@adityatelange 
Aren't robots.txt and robots meta tag supposed to have different purposes?

As per https://developers.google.com/search/docs/crawling-indexing/robots/intro

> While Google won't crawl or index the content blocked by a robots.txt file, we might still find and index a disallowed URL if it is linked from other places on the web. As a result, the URL address and, potentially, other publicly available information such as anchor text in links to the page can still appear in Google search results. To properly prevent your URL from appearing in Google search results, [password-protect the files on your server](https://developers.google.com/search/docs/crawling-indexing/control-what-you-share), [use the noindex meta tag or response header](https://developers.google.com/search/docs/crawling-indexing/block-indexing), or remove the page entirely.
